src/compiler/checker.ts

```diff
@@ -27133,8 +27133,9 @@ namespace ts {
                 }
             }
 
-            function isEvalNode(node: Expression) {
-                return node.kind === SyntaxKind.Identifier && (node as Identifier).escapedText === "eval";
+            function isInIndirectCall(node: BinaryExpression) {
+                return node.parent.kind === SyntaxKind.ParenthesizedExpression && (isCallExpression(node.parent.parent) && node.parent.parent.expression === node.parent || node.parent.parent.kind === SyntaxKind.TaggedTemplateExpression) &&
+                    (isAccessExpression(node.right) || isIdentifier(node.right) && node.right.escapedText === "eval");

```

Comment:
Why do we still have to check for `eval`? I think we want to silence the error for any indirect calls.